ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/answers/activity-text-input/activityTextInput.component.ts

```diff
@@ -1,185 +1,203 @@
 import {
-  Component,
-  EventEmitter,
-  Input,
-  OnChanges,
-  OnDestroy,
-  OnInit,
-  Output,
-  SimpleChanges,
+    Component,
+    EventEmitter,
+    Input,
+    OnChanges,
+    OnDestroy,
+    OnInit,
+    Output,
+    SimpleChanges,
 } from '@angular/core';
 import { FormControl, FormGroup } from '@angular/forms';
 import { Observable, of, Subject, Subscription } from 'rxjs';
-import { debounceTime, map, switchMap } from 'rxjs/operators';
+import { debounceTime, map, switchMap, tap } from 'rxjs/operators';
 
 import { ActivityTextQuestionBlock } from '../../../../models/activity/activityTextQuestionBlock';
 import { TextSuggestion } from '../../../../models/activity/textSuggestion';
 
 @Component({
-  selector: 'ddp-activity-text-input',
-  templateUrl: './activityTextInput.component.html',
-  styleUrls: ['./activityTextInput.component.scss'],
+    selector: 'ddp-activity-text-input',
+    templateUrl: './activityTextInput.component.html',
+    styleUrls: ['./activityTextInput.component.scss'],
 })
 export class ActivityTextInput implements OnInit, OnChanges, OnDestroy {
-  @Input() block: ActivityTextQuestionBlock;
-  @Input() placeholder: string;
-  @Input() readonly: boolean;
-  @Output() valueChanged = new EventEmitter<string | null>();
-  formGroup: FormGroup;
-  controlName: string;
-  confirmationControlName: string;
-  filteredSuggestions$: Observable<TextSuggestion[]>;
-  private sub: Subscription;
-  private input$ = new Subject<string>();
-  private readonly confirmationControlNamePrefix = 'confirmation_';
-  private readonly typeAheadMinInputLength = 3;
-
-  ngOnInit(): void {
-    this.buildForm();
-    this.initFilteredSuggestions();
-  }
-
-  ngOnChanges(changes: SimpleChanges): void {
-    if (changes.block && !changes.block.isFirstChange()) {
-      this.initFilteredSuggestions();
+    @Input() public block: ActivityTextQuestionBlock;
+    @Input() public placeholder: string;
+    @Input() public readonly: boolean;
+    @Output() public valueChanged = new EventEmitter<string | null>();
+    public formGroup: FormGroup;
+    public controlName: string;
+    public confirmationControlName: string;
+    public filteredSuggestions$: Observable<TextSuggestion[]>;
```

Comment:
We have similar functionality in the picklist component:
https://github.com/broadinstitute/ddp-angular/blob/d94522164cd36cf0bb1d15b6713f1e3b725d0568/ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/picklist/baseActivityPicklistQuestion.component.ts#L14

It might not be worthwhile to create a common counter component or even implement in same way, but YOU SHOULD reuse the CSS style used there and the translations (you see that the original implementation has entry for singular and plural form of the message).